src/fiber/context/x86_64.cr

```diff
@@ -3,39 +3,78 @@
 class Fiber
   # :nodoc:
   def makecontext(stack_ptr, fiber_main) : Nil
-    # in x86-64, the context switch push/pop 6 registers + the return address
-    # that is left on the stack, we thus reserve space for 7 pointers:
-    @context.stack_top = (stack_ptr - 7).as(Void*)
-    @context.resumable = 1
+    {% if flag?(:win32) %}
+      # In x86-64, the context switch push/pop 9 registers
+      @context.stack_top = (stack_ptr - 9).as(Void*)
+      @context.resumable = 1
 
-    stack_ptr[0] = fiber_main.pointer # %rbx: initial `resume` will `ret` to this address
-    stack_ptr[-1] = self.as(Void*)    # %rdi: puts `self` as first argument for `fiber_main`
+      stack_ptr[0] = fiber_main.pointer # %rbx: Initial `resume` will `ret` to this address
+      stack_ptr[-1] = self.as(Void*)    # %rcx: puts `self` as first argument for `fiber_main`
+    {% else %}
+      # in x86-64, the context switch push/pop 6 registers + the return address
+      # that is left on the stack, we thus reserve space for 7 pointers:
+      @context.stack_top = (stack_ptr - 7).as(Void*)
+      @context.resumable = 1
+
+      stack_ptr[0] = fiber_main.pointer # %rbx: initial `resume` will `ret` to this address
+      stack_ptr[-1] = self.as(Void*)    # %rdi: puts `self` as first argument for `fiber_main`
+    {% end %}
   end
 
   # :nodoc:
   @[NoInline]
   @[Naked]
   def self.swapcontext(current_context, new_context) : Nil
-    asm("
-      pushq %rdi        // push 1st argument (because of initial resume)
-      pushq %rbx        // push callee-saved registers on the stack
-      pushq %rbp
-      pushq %r12
-      pushq %r13
-      pushq %r14
-      pushq %r15
-      movq %rsp, 0($0)  // current_context.stack_top = %rsp
-      movl $$1, 8($0)   // current_context.resumable = 1
+    {% if flag?(:win32) %}
```

Comment:
Should we have a `x86_64-windows-msvc.cr` and a `x86_64-linux-gnu.cr`?

Reply:
Sounds reasonable.

Reply:
Why `x86_64-linux-gnu.cr`? The context switch is valid for all System-V targets (linux gnu/musl, openbsd, freebsd, ...). 

Are there alternatives to win32-msvc ABI, for example is MinGW different? In that case probably extract `x86_64-windows-msvc.cr` and have a`x86_64-posix.cr`.

Reply:
`x86_64-posix.cr` => `x86_64-unix.cr`? We're typically using `unix` everywhere else...

Reply:
Oh yeah, sure.

Reply:
I didn't know the standard abi had a name! If it's called systemv, then it should be named that!

`x86_64-sysv.cr` and `x86_64-microsoft.cr` would be my preference

Reply:
That naming looks good, see https://en.wikipedia.org/wiki/X86_calling_conventions#x86-64_calling_conventions